### ISSUE TITLE
[HEVC] Duration overflow issue.

### ISSUE THREAD

**Issue by redbaty** [radek-k/FFMediaToolkit#8]
First of all, I'd like to thank you for your efforts, this lib is pretty life saving (probably the first cross-platform library in .net which is not a complete mess).

Anyway, here's the problem, I've got an HEVC (H265) file which is exactly 22 minutes and 53 seconds long, but for some reason, the `StreamInfo` constructor fails to parse the correct duration causing an overflow (claiming that the duration is too long). I'm not so sure if this is related to this project or the project this is based on `FFmpeg.AutoGen`.

Here is some information that can be helpful:

* If I change line 32 on the file `StreamInfo.cs` to a fixed timespan, the library works as expected.
* The value for `stream->duration` is: -9223372036854775808
* The value for `stream->time_base` as double is 0.001

**Comment by redbaty** [comment]
I've found a workout, tho this issue might be weirder than I thought, should I provide a sample file?

**Comment by radek-k** [comment]
I've tested this library with a 30 minutes long 60 fps h265 file which has 108 000 frames(generated by the following command) and it's working correctly.
`ffmpeg -t 0:30:00 -f lavfi -i color=c=black:s=640x480 -c:v libx265 -x265-params crf=25 -r 60 -pix_fmt yuv420p c:/video-h265.mp4`

The bug may be in `FFmpeg.AutoGen` because `stream->duration` is set by it.
Can you provide a sample file?

**Comment by redbaty** [comment]
As on my pull request, for me, it only works when I take the duration from the `AVContext`.

[Heres the sample](https://drive.google.com/open?id=1WJHk_i0yg6VJIC2jdsr2G09cTUY-_AqY). (don't mind the anime).

**Comment by radek-k** [comment]
The bug is in `FFmpeg.AutoGen` or even in `ffmpeg`. Could you open a pull request with the workaround (and also with the chapters support - your[ 2e86dd8](https://github.com/redbaty/FFMediaToolkit/commit/2e86dd833015ef1b14c33c1e6581b5a7a8c8d725) commit)?

![devenv_ZRyeC743KW](https://user-images.githubusercontent.com/49524121/67146079-86590b00-f287-11e9-9872-a850ba40f046.png)

**Comment by redbaty** [comment]
Sure, I'll open some pull requests, the only thing I'm worried about is breaking another file in which the old `stream->duration` method works but the new `avcontext->duration` doesn't.

**Comment by radek-k** [comment]
Which file it will break?

**Comment by redbaty** [comment]
I mean, if we move from `stream->duration` to `AVContext->duration` there's a chance in some specific file it'll break, right?

**Comment by radek-k** [comment]
Stream duration is often equal to the file duration (converted to seconds), but in some cases it can be shorter. The library may get the stream duration from `AVContext` only if the `AVStream->duration` is unavailable.
Example:
````c#
if(stream->duration >= 0)
  return stream->duration;
else
  return avcontext->duration;
````

**Comment by redbaty** [comment]
Great suggestion! I've implemented it already. This way we keep compatibility and fixes for files which the stream duration is invalid.

**Comment by radek-k** [comment]
Fixed by #11
